test/gtest/kernel_tuning_net.cpp

```diff
@@ -0,0 +1,99 @@
+#include <miopen/miopen.h>
+
+#if MIOPEN_ENABLE_AI_HEUR
+#include "../tensor_holder.hpp"
+#include <gtest/gtest.h>
+#include <iostream>
+#include <miopen/convolution.hpp>
+#include <miopen/problem_description.hpp>
+#include <miopen/solver.hpp>
+#include "get_handle.hpp"
+#endif
+
+void Test_908_ConvAsm1x1U(void)
+{
+#if MIOPEN_ENABLE_AI_HEUR
```

Comment:
Jehandad wants to remove the dependency of the tests on MIOpen being built. This means MIOpen environment variables should not affect how the tests run. I have re-worked my tests so that they only depend on the function RunParameterPredictionModel() which does not depend on this environment variable being set.

Reply:
@Dmantri98 Please clarify: _Jehandad wants to remove the dependency of the tests on HOW MIOpen IS being built_, right? In other words, build options should not affect the tests?

But `MIOPEN_ENABLE_AI_HEUR` macro is an build option, while environment variable is not. The env var can be considered as simply another internal API of the library that unit test is allowed to use.

Reply:
If we use the solver `HeuristicInit` to test the heuristic, then we would have to force the env var in the test to make sure that the code path in MIOpen is enabled. However, that did not seem an elegant design to me. Therefore, @Dmantri98  has restructured the code so that the function under test works regardless of the environment. I hope this resolves this conversation.

Reply:
Resolved